ADBDEV-7239-1: Resolve conflicts in heapam.c

KnightMurloc opened the pull request:
The signature of the HeapTupleSatisfiesUpdate function is slightly different
in gpdb, which led to a conflict with cab72f0fd08cceb975194d2acd2f6eed917e8fb8

RekGRpth commented:
Why did you put unrelated changes into one PR?

RekGRpth commented on src/backend/access/transam/xlog.c:
```diff
@@ -2155,13 +2155,7 @@ AdvanceXLInsertBuffer(XLogRecPtr upto, bool opportunistic)
-#ifdef WAL_DEBUG
-	int			npages = 0;
-#endif
-=======
 	int			npages pg_attribute_unused() = 0;
->>>>>>> REL_12_13
```
There may be more similar conflict areas.

KnightMurloc committed:
Resolve conflicts heapam.c

The signature of the HeapTupleSatisfiesUpdate function is slightly different
in gpdb, which led to a conflict with cab72f0fd08cceb975194d2acd2f6eed917e8fb8
